packages/neon-core/src/sc/manifest/ContractPermission.ts

```diff
@@ -0,0 +1,49 @@
+export interface ContractPermissionLike {
+  contract: string;
+  methods: "*" | string[];
+}
+
+export class ContractPermission {
+  public contract: string;
+  public methods: "*" | string[];
+
+  public constructor(obj: Partial<ContractPermissionLike> = {}) {
+    const { contract = "*", methods = "*" } = obj;
+    this.contract = this._formatContract(contract);
+    this.methods = methods;
+  }
+
+  public get isHash(): boolean {
+    return this.contract.length === 40;
+  }
+
+  public get isGroup(): boolean {
+    return this.contract.length === 66;
+  }
+
+  public get isWildcard(): boolean {
+    return this.contract === "*";
+  }
+
+  private _formatContract(contract: string): string {
+    switch (true) {
+      case contract.length === 66: // public key of contract manifest group
+      case contract.length === 40: // contract scripthash
+      case contract === "*": // wildcard, means it accept any contract
+        return contract;
+      case contract.length === 42 && contract.indexOf("0x") === 0: // contract with "0x" prefix
+        return contract.slice(2);
+      default:
+        throw new Error(
+          `This is not a ContractPermissionDescriptor: ${contract}`
+        );
+    }
+  }
+
+  public export(): ContractPermissionLike {
+    return {
+      contract: this.isHash ? `0x${this.contract}` : this.contract,
```

Comment:
When did we ever export with `0x` appended? Its inconsistent so lets not use it. It just means the string is a hexadecimal string. In fact if we are supporting it fully, we should be checking it for the public key too (since its also a hexadecimal string). I would remove all 0x support at the moment because we did not start with support and adding it right now is a mess.

Reply:
The `export` function of Manifest should be able to be used for contract deployment, I added `0x` just in case it cannot.

Reply:
And why would the manifest require `0x` for hash but not public key? I think we should remove this as it is inconsistent at the moment.

Reply:
OK, changed